n2survey/lime/transformations.py

```diff
@@ -231,3 +229,30 @@ def rate_mental_health(
         df = df.drop(df.columns[:-2], axis=1)
 
     return df
+
+
+def calculate_phd_duration(start_responses: pd.DataFrame, end_responses: pd.DataFrame):
+    """Calculate PhD duration
+    Args:
+        start_responses (pd.DataFrame): DataFrame containing responses data of the start of duration
+        end_responses (pd.DataFrame): DataFrame containing responses data of the end of duration
+    Returns:
+        pd.DataFrame: Rounded PhD duration in days, months and years"""
+
+    # get labels of start and end questions to get their data
+    df = pd.concat(
+        [start_responses, end_responses],
+        axis=1,
+    )
+
+    # duration calculation, return as day
+    df["PhD duration (days)"] = df.iloc[:, 1] - df.iloc[:, 0]
+
+    # convert days to month and year
+    df["PhD duration (months)"] = df.iloc[:, 2] / np.timedelta64("1", "M")
+    df["PhD duration (years)"] = df.iloc[:, 2] / np.timedelta64("1", "Y")
```

Comment:
The resulting DataFrame has dtypes as follows:
![image](https://user-images.githubusercontent.com/77344869/168490368-fd832fcc-1028-44e1-9782-33241f3d5252.png)

This inconsistency will likely cause problems in downstream tasks. It's probably better to set all to float.

Reply:
Hey Dawei,
Please check the latest commit that I made. I agree with you that the columns should have consistent data type but because there are NaTs or NaNs, I changed the `PhD duration (days)` to float instead, is that okay?

Reply:
A test failed because in Line 253 you are dividing a `float64` dtype by a `<m8[ns]` dtype. I suggest you convert dtypes at the end after all other operations.

Reply:
I forgot to change the test according to the new dtype, I just changed the test to accommodate to the new dtype. I should work now, do you think that's okay?

Reply:
Yes, looks fine now.